src/drivers/gnss/septentrio/septentrio.cpp

```diff
@@ -0,0 +1,1439 @@
+
+	case SBF_ID_VelCovGeodetic: SBF_TRACE_RXMSG("Rx VelCovGeodetic");
+		_msg_status |= 2;
+		_report_gps_pos.s_variance_m_s = _buf.payload_vel_col_geodetic.cov_ve_ve;
+
+		if (_report_gps_pos.s_variance_m_s < _buf.payload_vel_col_geodetic.cov_vn_vn) {
+			_report_gps_pos.s_variance_m_s = _buf.payload_vel_col_geodetic.cov_vn_vn;
+		}
+
+		if (_report_gps_pos.s_variance_m_s < _buf.payload_vel_col_geodetic.cov_vu_vu) {
+			_report_gps_pos.s_variance_m_s = _buf.payload_vel_col_geodetic.cov_vu_vu;
+		}
+
+		//SBF_DEBUG("VelCovGeodetic handled");
+		break;
+
+	case SBF_ID_DOP: SBF_TRACE_RXMSG("Rx DOP");
+		_msg_status |= 4;
+		_report_gps_pos.hdop = _buf.payload_dop.hDOP * 0.01f;
+		_report_gps_pos.vdop = _buf.payload_dop.vDOP * 0.01f;
+		//SBF_DEBUG("DOP handled");
+		break;
+
+	case SBF_ID_AttEuler: SBF_TRACE_RXMSG("Rx AttEuler");
+
+		if (!_buf.payload_att_euler.error_not_requested) {
+
+			int error_aux1 = _buf.payload_att_euler.error_aux1;
+			int error_aux2 = _buf.payload_att_euler.error_aux2;
+
+			// SBF_DEBUG("Mode: %u", _buf.payload_att_euler.mode)
+			if (error_aux1 == 0 && error_aux2 == 0) {
+				float heading = _buf.payload_att_euler.heading;
+				heading *= M_PI_F / 180.0f; // deg to rad, now in range [0, 2pi]
+
+
+				if (heading > M_PI_F) {
+					heading -= 2.f * M_PI_F; // final range is [-pi, pi]
+				}
+
+				_report_gps_pos.heading = heading;
+				// SBF_DEBUG("Heading: %.3f rad", (double) _report_gps_pos.heading)
+				//SBF_DEBUG("AttEuler handled");
+
+			} else if (error_aux1 != 0) {
+				//SBF_DEBUG("Error code for Main-Aux1 baseline: Not enough measurements");
+			} else if (error_aux2 != 0) {
+				//SBF_DEBUG("Error code for Main-Aux2 baseline: Not enough measurements");
+			}
+		} else {
+			//SBF_DEBUG("AttEuler: attitude not requested by user");
+		}
+
+
+		break;
+
+	case SBF_ID_AttCovEuler: SBF_TRACE_RXMSG("Rx AttCovEuler");
+
+		if (!_buf.payload_att_cov_euler.error_not_requested) {
+			int error_aux1 = _buf.payload_att_cov_euler.error_aux1;
+			int error_aux2 = _buf.payload_att_cov_euler.error_aux2;
+
+			if (error_aux1 == 0 && error_aux2 == 0) {
+				float heading_acc = _buf.payload_att_cov_euler.cov_headhead;
+				heading_acc *= M_PI_F / 180.0f; // deg to rad, now in range [0, 2pi]
+				_report_gps_pos.heading_accuracy = heading_acc;
+				// SBF_DEBUG("Heading-Accuracy: %.3f rad", (double) _report_gps_pos.heading_accuracy)
+				//SBF_DEBUG("AttCovEuler handled");
+
+			} else if (error_aux1 != 0) {
+				//SBF_DEBUG("Error code for Main-Aux1 baseline: %u: Not enough measurements", error_aux1);
+			} else if (error_aux2 != 0) {
+				//SBF_DEBUG("Error code for Main-Aux2 baseline: %u: Not enough measurements", error_aux2);
+			}
+		} else {
+			//SBF_DEBUG("AttCovEuler: attitude not requested by user");
+		}
+
+		break;
+
+	default:
+		break;
+	}
+
+	if (ret > 0) {
+		// NOTE: Isn't this always 0?
+		_report_gps_pos.timestamp_time_relative = static_cast<int32_t>(_last_timestamp_time - _report_gps_pos.timestamp);
```

Comment:
Seems to always be 0 in the MAVLink inspector. I don't fully understand what the purpose of this line is as `_last_timestamp_time` is set to `_report_gps_pos.timestamp` a bit before this line, so it's always 0.